### PR TITLE
add multiplechoice filter for category in featured program

### PR DESCRIPTION
**Pull request by rachadiannovansyah**
Overview:
- Add multiplechoice for filter category featured program
- so FE can send params like this `{base_url}/v1/featured-params?cat[]=kesehatan&cat[]=pendidikan`
that's params has been discussed and accepted by Frontend team

@jabardigitalservice/jds-backend 

Evidence:
project: Portal Jabar
title: Menambahkan multiple filter kategori pada Program Unggulan
participants: @rachadiannovansyah @setiadijoe @ayocodingit @khihadysucahyo

### CODE REVIEW

**Review by setiadijoe — APPROVED** [review]

**Review by ayocodingit — APPROVED** [review]

File: core-service/src/domain/featured_program.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ type FeaturedProgram struct {
 	Excerpt      string           `json:"excerpt"`
 	Description  string           `json:"description"`
 	Organization string           `json:"organization"`
-	Categories   JSONStringSlices `json:"categories"`
+	Categories   string           `json:"categories"`
```
Comment (khihadysucahyo) [inline core-service/src/domain/featured_program.go:15]: use string as type for field categories(?)
is the payload still an array or not?

File: core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go [hunk 2]
```diff
@@ -63,9 +63,17 @@ func (m *mysqlFeaturedProgramRepository) Fetch(ctx context.Context, params *doma
...
+	for idx, cat := range data {
+		if v, ok := params.Filters["categories"]; ok && v != "" {
+			if idx == 0 {
+				query = fmt.Sprintf(`%s AND json_search(categories, 'all', '%s') is not null`, query, cat)
+			} else {
+				query = fmt.Sprintf(`%s OR json_search(categories, 'all', '%s') is not null`, query, cat)
+			}
+		}
 	}
+	query = query + ` limit 50`
```
Comment (khihadysucahyo) [inline core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go:76]: should we consider making the sql syntax consistent. uppercase? (e.g `FROM, LIMIT, JSON_SEARCH`)

File: core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go [hunk 3]
```diff
@@ -63,9 +63,17 @@ func (m *mysqlFeaturedProgramRepository) Fetch(ctx context.Context, params *doma
...
 	query := `SELECT id, title, excerpt, description, organization, categories, service_type, websites, social_media, logo FROM featured_programs WHERE 1=1`
 
-	if v, ok := params.Filters["categories"]; ok && v != "" {
-		query = fmt.Sprintf(`%s AND json_search(categories, 'all', '%s') is not null limit 50`, query, v)
+	data := params.Filters["categories"].([]string)
+	for idx, cat := range data {
+		if v, ok := params.Filters["categories"]; ok && v != "" {
+			if idx == 0 {
+				query = fmt.Sprintf(`%s AND json_search(categories, 'all', '%s') is not null`, query, cat)
+			} else {
```
Comment (khihadysucahyo) [inline core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go:71]: i was just wondering how to simplify this ifelse query.. and i think we should consider using brackets AND (json search, OR ...), to make sure this query can be added with other query conditions in the future

File: core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go (hunk 2 above)
Comment (rachadiannovansyah) [reply core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go]: check out here! https://github.com/jabardigitalservice/portal-jabar-services/pull/67/commits/6b0c7d2375c80e7b86d0082855ab51db547bc43c

Comment (khihadysucahyo) [reply core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go]: your latest commit is missing this, dude 🙃
 `IS NOT NULL`

File: core-service/src/domain/featured_program.go (hunk 1 above)
Comment (khihadysucahyo) [reply core-service/src/domain/featured_program.go:15]: gak ada oyy wkwk

Comment (rachadiannovansyah) [reply core-service/src/domain/featured_program.go]: disini oy blm ke push wkakaka https://github.com/jabardigitalservice/portal-jabar-services/pull/67/commits/b04b335758db96e03a6cdc7f4aa648b7f21c9bc0

File: core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go (hunk 2 above)
Comment (rachadiannovansyah) [reply core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go]: updated here gans! https://github.com/jabardigitalservice/portal-jabar-services/pull/67/commits/185ffa6ed731388c2cc012c0dc8c0d89e766c555

Comment (khihadysucahyo) [reply core-service/src/modules/featured-program/repository/mysql/mysql_featured_program.go]: sapss dahlah ku lanjut cuti aja wkwk

**Review by khihadysucahyo — APPROVED** [review]

**Pull request merged**